src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/ReleaseDataFileServiceTest.cs

```diff
@@ -201,16 +221,8 @@ await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContex
                 var result = await service.Delete(release.Id, releaseDataFile.ReleaseFileReference.Id);
 
                 Assert.True(result.IsRight);
-
-                blobStorageService.Verify(mock =>
-                    mock.DeleteBlob(PrivateFilesContainerName,
-                        AdminReleasePath(release.Id, ReleaseFileTypes.Data, "Data 1.csv")), Times.Once());
-                blobStorageService.Verify(mock =>
-                    mock.DeleteBlob(PrivateFilesContainerName,
-                        AdminReleasePath(release.Id, ReleaseFileTypes.Metadata, "Data 1.meta.csv")), Times.Once());
-
-                importService.Verify(mock =>
-                    mock.RemoveImportTableRowIfExists(release.Id, "Data 1.csv"), Times.Once());
+                
+                MockUtils.VerifyAllMocks(blobStorageService, importService);
```

Comment:
On this branch I have set up all the mocks to Strict behaviour so the test automatically fails if there's any calls set up that are not used, or any unexpected invocations, without needing to `VerifyAll` and `VerifyNoOtherCalls` every time. I think this was already the case for at least some of the tests in this class.

https://github.com/dfe-analytical-services/explore-education-statistics/compare/EES-1560

Reply:
I think it's still valuable to call VerifyAll() because of a very real case in this particular test class.

VerifyAll() helps because:

Strict mode ensures that any invocations on the mock without corresponding Setups fail, which is great - it means you don't need to manually Verify() those setups afterwards, hence why I've removed a bunch of manual Verifys in this test class in test methods that use Strict mocks

However, it does not help you spot false positives, like Setups() that never get invoked.  In this test class for instance there is a test called `DeleteAll_FileFromAmendment()` which was Setting Up:

```
importService.Setup(mock => mock.RemoveImportTableRowIfExists(release.Id, "data.csv"))	            blobStorageService
                .Returns(Task.CompletedTask);	                .Setup(mock =>

                    mock.ListBlobs(PrivateFilesContainerName, AdminReleaseBatchesDirectoryPath(amendmentRelease.Id)))
            blobStorageService.Setup(mock => mock.DeleteBlob(PrivateFilesContainerName, It.IsAny<string>()))	                .ReturnsAsync(new List<BlobInfo>());
                .Returns(Task.CompletedTask);
```

However, neither of these Setups were actually getting called when running the test, explicitly because when deleting a File from an Amendment, you do NOT want to remove the import table row.

The VerifyAll() ensures that all Setups were called, and hence it caught out this test's incorrect Setup assumptions.

Reply:
I'm sure I'd seen tests fail where I'd setup behaviour that wasn't used, but it turns out not. Thanks for explaining!